dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -555,4 +656,24 @@ long languageId () {
             return (long)this.getKey(1);
         }
     }
+
+    private static class VanityMatches {
+
+        private final boolean patternMatches;
+        private final String  [] groups;
+
+        public VanityMatches(final boolean patternMatches,
+                             final String[] groups) {
+            this.patternMatches = patternMatches;
+            this.groups = groups;
+        }
+
+        boolean isPatternMatches() {
+            return patternMatches;
+        }
+
+        String[] getGroups() {
+            return groups;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "groups". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)